enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,628 @@
+---
+title: Cluster Logging Loki as an alternative Log Store
+authors:
+  - "@periklis"
+reviewers:
+  - "@ewolinetz"
+  - "@blockloop"
+  - "@lukas-vlcek"
+  - "@sichvoge"
+approvers:
+  - "@ewolinetz"
+  - "@alanconway"
+  - "@jcantril"
+creation-date: 2021-03-29
+last-updated: 2021-03-29
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Cluster Logging: Loki as an alternative Log Store
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of this is to provide an alternative low-retention log store based on [Loki](https://github.com/grafana/loki) for the cluster logging stack. This aims to provide a cloud-native first solution specialized in storing logs and in turn lessen the operational cost running cluster logging.
+
+Currently our main log store, i.e. Elasticsearch, is a more general-purpose ingestion, indexing and querying cluster-based solution for text search. It is a powerful solution with fine-grained control on how to scale log storage, indexing and querying.
+
+In general it serves its purpose quite well but it is a constant source of labor regarding operations. Its powerfulness and complexity requires a constant toll on the human operator side to understand how to individually scale the cluster as a log store. Representative actions that a human operator needs to figure out besides controlled operations by [elasticsearch operator](https://github.com/openshift/elasticsearch-operator):
+- Regular evaluation and balance of the CPU/Memory requests and limits because configuration for ingestion and query paths is not individual configurable.
+- Regular evaluation and adaption of redundancy effects on memory and persistent volume usage.
+- Usage of `managementState: Unmanaged` to unblock the cluster in situations of bugs, e.g. locked indices, shard allocation.
+- Usage of the Elasticsearch cluster API to tune the cluster for options not supported in the Elasticearch custom resource.
```

Comment:
Elasticsearch cluster

Reply:
Nope `custom resource` is correct here. The sentence is dedicated to user poking directly on Elastricsearch via its Cluster API because we don't expose all knobs in the Elasticearch custom resource definition.